From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Josef Holzmayr <jester@theyoctojester.info>
Date: Fri, 25 Apr 2026 06:00:00 +0000
Subject: [PATCH] imx93_11x11_frdm: Enable Mender configuration

Enable redundant environment storage in MMC and the bootcount framework
so meta-mender can install its bootloader environment and perform safe
A/B updates with rollback on the i.MX 93 11x11 FRDM board.

Targets NXP downstream u-boot-imx lf_v2025.04 as shipped with the
whinlatter (LF 6.18.2) BSP.

Upstream-Status: Inappropriate [Mender configuration]

Signed-off-by: Josef Holzmayr <jester@theyoctojester.info>
---
 configs/imx93_11x11_frdm_defconfig | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/configs/imx93_11x11_frdm_defconfig b/configs/imx93_11x11_frdm_defconfig
--- a/configs/imx93_11x11_frdm_defconfig
+++ b/configs/imx93_11x11_frdm_defconfig
@@ -7,7 +7,7 @@ CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_SF_DEFAULT_SPEED=40000000
-CONFIG_ENV_SIZE=0x4000
+CONFIG_ENV_SIZE=0x20000
 CONFIG_ENV_OFFSET=0x700000
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="imx93-11x11-frdm"
@@ -24,6 +24,7 @@ CONFIG_SPL_HAS_BSS_LINKER_SECTION=y
 CONFIG_SPL_BSS_MAX_SIZE=0x2000
 CONFIG_SYS_LOAD_ADDR=0x80400000
 CONFIG_SPL=y
+CONFIG_ENV_OFFSET_REDUND=0x740000
 CONFIG_CMD_DEKBLOB=y
 CONFIG_SPL_IMX_ROMAPI_LOADADDR=0x88000000
 CONFIG_SYS_MEMTEST_START=0x80000000
@@ -84,8 +85,8 @@ CONFIG_CMD_EXT4_WRITE=y
 CONFIG_OF_CONTROL=y
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_NOWHERE=y
 CONFIG_ENV_IS_IN_MMC=y
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
 CONFIG_SYS_MMC_ENV_DEV=1
 CONFIG_USE_ETHPRIME=y
@@ -95,6 +96,8 @@ CONFIG_SPL_DM=y
 CONFIG_SYSCON=y
 CONFIG_ADC=y
 CONFIG_ADC_IMX93=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
 CONFIG_SPL_CLK_IMX93=y
 CONFIG_CLK_IMX93=y
 CONFIG_CPU=y
--
2.40.1

